### PR TITLE
Guzman-Neilan H1(div) macroelement

### PR DESCRIPTION
**Pull request by pbrubeck**
Depends on https://github.com/firedrakeproject/fiat/pull/86

### CODE REVIEW

File: finat/guzman_neilan.py [hunk 1]
```diff
@@ -8,18 +8,28 @@ class GuzmanNeilan(PiolaBubbleElement):
...
-    def __init__(self, cell, degree=None, subdegree=1):
+    def __init__(self, cell, order=1):
         if Citations is not None:
             Citations().register("GuzmanNeilan2019")
-        super().__init__(FIAT.GuzmanNeilan(cell, degree=degree, subdegree=subdegree))
+        super().__init__(FIAT.GuzmanNeilan(cell, order=order))
+
+
+class GuzmanNeilanSecondKind(PiolaBubbleElement):
+    """H1-conforming macroelement construced as the enrichment of C0 Pk^d(Alfeld)
```
Comment (wence-) [inline finat/guzman_neilan.py:18]:
```suggestion
    """H1-conforming macroelement constructed as the enrichment of C0 Pk^d(Alfeld)
```

File: finat/guzman_neilan.py [hunk 2]
```diff
@@ -5,15 +5,40 @@
 
 
 class GuzmanNeilan(PiolaBubbleElement):
-    def __init__(self, cell, degree=None, subdegree=1):
-        sd = cell.get_spatial_dimension()
-        if degree is None:
-            degree = sd
+    """H1-conforming macroelement construced as the enrichment of Pk^d
+    with Guzman-Neilan facet bubbles.
```
Comment (rckirby) [inline finat/guzman_neilan.py:9]: Does this comment fit on one line?

File: finat/piola_mapped.py [hunk 3]
```diff
@@ -125,20 +127,30 @@ def basis_transformation(self, coordinate_mapping):
         detJ = coordinate_mapping.detJ_at(bary)
 
         dofs = self.entity_dofs()
-        edofs = self._element.entity_dofs()
+        bfs = self._element.entity_dofs()
```
Comment (rckirby) [inline finat/piola_mapped.py:130]: Is `bfs` a good label -- the `entity_dofs` identifies degrees of freedom, which I guess are in exact correspondence to basis functions.

**Review by rckirby — APPROVED** [review]
Looks good to me.

File: finat/piola_mapped.py (hunk 3 above)
Comment (pbrubeck) [reply finat/piola_mapped.py:130]: `edofs` stood for extended dofs, now here it is useful to think that we have more basis functions than degrees of freedom, and that V.T maps basis functions of the extended reference space to physical nodal basis functions.

**Review by rckirby — APPROVED** [review]
Up to deciding on whether to add "FirstKind", I think this is fine.

File: finat/ufl/elementlist.py [hunk 4]
```diff
@@ -133,13 +133,17 @@ def show_elements():
...
+
 register_element("Arnold-Qin", "AQ", 1, H1, "identity", (2, 2), ("triangle",))
 register_element("Reduced-Arnold-Qin", "AQ-red", 1, H1, "contravariant Piola", (2, 2), ("triangle",))
 register_element("Christiansen-Hu", "CH", 1, H1, "contravariant Piola", (1, 1), simplices[1:])
-register_element("Guzman-Neilan", "GN", 1, H1, "contravariant Piola", (2, 3), simplices[1:])
-register_element("Guzman-Neilan Bubble", "GNB", 1, H1, "contravariant Piola", (2, 3), simplices[1:])
 register_element("Alfeld-Sorokina", "AS", 1, H1, "contravariant Piola", (2, 2), simplices[1:])
-register_element("Johnson-Mercier", "JM", 2, HDivDiv, "double contravariant Piola", (1, 1), simplices[1:])
+
+register_element("Guzman-Neilan", "GN", 1, H1, "contravariant Piola", (1, None), simplices[1:])
```
[the remark this answers is not in the document]
Comment (pbrubeck) [reply finat/ufl/elementlist.py]: Done. `"Guzman-Neilan 1st kind H1"` and `"Guzman-Neilan 2nd kind H1"`

**Pull request merged**